mcs/class/Microsoft.Build.Utilities/Microsoft.Build.Utilities/ProcessWrapper.cs

```diff
@@ -74,6 +74,7 @@ public new void Start ()
 		public void WaitForOutput (int milliseconds)
 		{
 			CheckDisposed ();
+			WaitForExit (milliseconds);
 			WaitHandle.WaitAll (new WaitHandle[] { endEventOut, endEventErr, endEventExit }, milliseconds);
```

Comment:
I would do something more like:
```
if (WaitHandle.WaitAll (new WaitHandle[] { endEventOut, endEventErr, endEventExit }, milliseconds))
  WaitForExit ();
```
What you did here might wait for `milliseconds` twice, and if `WaitForExit(milliseconds)` returns false, `endEventOut`, `endEventErr` and `endEventExit` will very likely be not signalled yet.